app/scripts/background.js

```diff
@@ -102,6 +104,15 @@ const initApp = async (remotePort) => {
 
 if (isManifestV3) {
   browser.runtime.onConnect.addListener(initApp);
+  // Message below signals content script in DAPPS to connect to metamask background as backend is not active
+  // It is required to re-connect DAPPS after service worker re-activation
+  chrome.tabs.query({ active: true, currentWindow: true }, function (tabs) {
+    chrome.tabs.sendMessage(
+      tabs[0].id,
+      { name: 'METAMASK_EXTENSION_READY' },
```

Comment:
Would `'METAMASK_EXTENSION_READY'` be helpful as a shared constant?

Reply:
@ryanml does this work?

Reply:
@ryanml does this work? [b9fdffa](https://github.com/MetaMask/metamask-extension/pull/16075/commits/b9fdffa6ac0c3ef4a118b0daeb804e9ea5b50a9b)

Reply:
Other messages we could add into this new constant: 
- `WORKER_KEEP_ALIVE_MESSAGE`
- `MESSAGE_TEXT` 

to do in follow-up PR

Reply:
works for me! thank you :)

Reply:
sweet :) np! Thanks for asking @ryanml